packages/build/src/barque.ts

```diff
@@ -0,0 +1,137 @@
+  Redhat = 'x86_64',
+}
+
+export class Barque {
+  private config: Config;
+  private mongodbEdition: string;
+  private mongodbVersion: string;
+  private arch: string;
+
+  constructor(config: Config) {
+    this.config = config;
+    // hard code mongodb edition to 'org' for now
+    this.mongodbEdition = 'org';
+    // linux mongodb versions to release to. This should perhaps be an array of
+    // [4.3.0, 4.4.0], like mongo-tools
+    this.mongodbVersion = '4.4.0';
+  }
+
+  /**
+   * Upload current package to barque, MongoDB's PPA for linux distros.
+   *
+   * @param {string} tarballURL- The uploaded to Evergreen tarball URL.
+   * @param {Config} config - Config object.
+   *
+   * @returns {Promise} The promise.
+   */
+  async releaseToBarque(tarballURL: string): Promise<any> {
+    const repoConfig = path.join(this.config.rootDir, 'config', 'repo-config.yml');
+
+    if (this.config.platform === Platform.Linux) {
+      try {
+        await this.execCurator(tarballURL, repoConfig);
+      } catch (error) {
+        throw new Error(`Curator is unable to upload to barque ${error}`);
+      }
+    }
+
+    return;
+  }
+
+  /**
+   * Run the child_process.exec to run curator command.
+   *
+   * @param {string} tarballURL- The uploaded to Evergreen tarball URL.
+   * @param {string} repoConfig - Path to repo-config.yml used to uplaod assets
+   * to appropriate distros.
+   *
+   * @returns {Promise} The promise.
+   */
+  async execCurator(
+    tarballURL: string,
+    repoConfig: string): Promise<any> {
+    return await execFile(
+      'curator', [
+        '--level', 'debug',
+        'repo', 'submit',
+        '--service', 'https://barque.corp.mongodb.com',
+        '--config', repoConfig,
+        '--distro', this.determineDistro(this.config.buildVariant),
+        '--arch', this.determineArch(this.config.buildVariant),
+        '--edition', this.mongodbEdition,
+        '--version', this.mongodbVersion,
+        '--packages', tarballURL
+      ], {
+      // curator looks for these options in env
+        env: {
+          NOTARY_KEY_NAME: 'server-4.4',
+          NOTARY_TOKEN: process.env.NOTARY_TOKEN,
+          BARQUE_API_KEY: process.env.BARQUE_API_KEY,
+          BARQUE_USERNAME: process.env.BARQUE_USERNAME
+        }
+      });
+  }
+
+  /**
+   * Determine the current arch to be passed on to curator given current build
+   * variant.
+   *
+   * @param {string} variant - Current build variant.
+   *
+   * @returns {string} Arch to be passed as an argument to curator
+   */
+  determineArch(variant: string): string {
+    // we can't use distro_id from evergreen's env variables, since those
+    // sometimes come with other attributes like -test -large -small, and are not
+    // valid distros for barque.
+    if (variant === BuildVariant.Linux) return Arch.Ubuntu;
+    if (variant === BuildVariant.Debian) return Arch.Debian;
+    if (variant === BuildVariant.Redhat) return Arch.Redhat;
+
+    return Distro.Ubuntu;
```

Comment:
Is it intentional that other returns are `Arch` and this one is `Distro`?

Reply:
nope, not intentional! thanks